roles/iam-user/tasks/create_users.yml

```diff
@@ -0,0 +1,26 @@
+
+---
+- name: Create IAM users
+  iam:
+    state: present
+    iam_type: user
+    name: "{{ item.name }}"
+    groups: "{{ item.groups }}"
```

Comment:
I think the problem is that a user could be a member or multiple groups so the loop need to process all the items and the groups list from the item.